specs/lightclient/validator.md

```diff
@@ -0,0 +1,395 @@
+*NOTE*: (TODO) details of when to join subnets
+
+#### Sync committee contributions
+
+Each slot some sync committee members are selected to aggregate the `SyncCommitteeSignature`s into a `SyncCommitteeContribution` which is broadcast on a global channel for inclusion into the next block.
+
+##### Aggregation selection
+
+A validator is selected to aggregate based on the computation in `is_sync_committee_aggregator` where `state` is a `BeaconState` with the relevant sync committees for the queried `slot`, i.e. `state.slot` is within the span covered by the current or next sync committee period. `privkey` is the BLS private key associated with the validator.
+
+```python
+def get_sync_committee_slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(
+      state, 
+      DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF, 
+      compute_epoch_at_slot(slot)
+    )
+    signing_root = compute_signing_root(slot, domain)
+    return bls.Sign(privkey, signing_root)
+
+def is_sync_committee_aggregator(state: BeaconState, slot: Slot, privkey: int) -> bool:
+    signature = get_sync_committee_slot_signature(state, slot, privkey)
+    modulo = max(
+      1, 
+      SYNC_COMMITTEE_SIZE // TARGET_AGGREGATORS_PER_SYNC_COMMITTEE
+    )
+    return bytes_to_uint64(hash(signature)[0:8]) % modulo == 0
+```
+
+*NOTE*: the set of aggregators will generally change every slot; however, the assignments can be computed ahead of time as soon as the committee is known
+
+##### Construct sync committee contribution
+
+If a validator is selected to aggregate the `SyncCommitteeSignature`s produced on a subnet during a given `slot`, they construct an aggregated `SyncCommitteeContribution`. Given all of the collected `sync_committee_signatures` from the `sync_committee_{subnet_id}` gossip, create the contribution via `construct_sync_committee_contribution`. `state` in this function is the same `BeaconState` used by the aggregator to produce their individual signature for the assigned slot.
+
+```python
+def construct_sync_committee_contribution(state: BeaconState, sync_committee_signatures: Set[SyncCommitteeSignature]) -> SyncCommitteeContribution:
+    contribution = SyncCommitteeContribution()
+    contribution.slot = state.slot
+
+    sync_committee = state.current_sync_committee
+    epoch = compute_epoch_at_slot(state.slot)
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, epoch)
+    signing_root = compute_signing_root(
+      get_block_root_at_slot(state, state.slot), 
+      domain
+    )
+
+    signatures = []
+    for sync_committee_signature in sync_committee_signatures:
+        index = sync_committee_signature.index
+        signature = sync_committee_signature.signature
+        contribution.aggregate_bits[index] = True
+        signatures.append(signature)
+
+        # is sync committee signature valid?
+        assert sync_committee_signature.slot == state.slot
+        assert bls.Verify(
```

Comment:
I don't think we need assertions at this point in the spec. The conditions that these values would be passed into this function would be that they were verified during gossip validations. I would just note that you collected the valid `SyncCommitteeSignature`s for the slot and leave the validations out

Reply:
will remove as i add gossip conditions